scIB/metrics.py

```diff
@@ -302,14 +309,24 @@ def ari(adata, group1, group2):
     return sklearn.metrics.cluster.adjusted_rand_score(group1, group2)
 
 ### Isolated label score
-def isolated_labels(adata, label_key, batch_key, cluster_key="iso_cluster", 
-                    cluster=True, n=None, all_=False, verbose=True):
+def isolated_labels(
+        adata,
+        label_key,
+        batch_key,
+        embed,
+        cluster=True,
+        n=None,
+        all_=False,
+        verbose=True
+):
     """
     score how well labels of isolated labels are distiguished in the dataset by
         1. clustering-based approach
         2. silhouette score
     params:
         cluster: if True, use clustering approach, otherwise use silhouette score approach
+        embed: key in adata.obsm used for silhouette score if cluster=True or
+            as representation for clustering (if neigbors missing in adata)
```

Comment:
In our current pipeline reduce_data is always called though, so the F1 score never had an issue, right? The addition to neighbourhood graphs for cluster-based iso label scoring is a good fall-back, but it was never an issue, no?

Reply:
Good point. The clustering for the F1 isolated label score is independent from the global clustering used by ARI, NMI etc., called by `reduce_data`. The latter case was using the correct embedding, even when `recompute_pca` was erroneously set to `True` for embedding output. However, the isolated label F1 score would have been computed on the `X_pca` instead of the actual embedding, so that we would expect different results.

Reply:
Correct, `use_rep` is only used if `adata.uns[neighbors]` is missing. In cases where `reduce_data` is not run, that'll ensure that the neighborhood graph is computed on the correct representation.